File: fixtures/scale/php/large/src/Families/Sum16/Sum16Descriptor.php
<?php

namespace CipherLab\Families\Sum16;

use CipherLab\Abstractions\Cipher;
use CipherLab\Abstractions\FamilyDescriptor;

// Registered once in FamilyCatalog; ties together Sum16Cipher and
// Sum16Vectors under the "hash" suite.
class Sum16Descriptor implements FamilyDescriptor
{
    public function family(): string
    {
        return 'sum16';
    }

    public function suite(): string
    {
        return 'hash';
    }

    public function cipher(): Cipher
    {
        return new Sum16Cipher();
    }

    public function vectors(): array
    {
        return Sum16Vectors::all();
    }
}

File: fixtures/scale/objc/large/Families/Sum16/Sum16Descriptor.m
#import "Sum16Descriptor.h"
#import "Sum16Cipher.h"
#import "Sum16Vectors.h"

@implementation Sum16Descriptor

- (NSString *)family {
    return @"sum16";
}

- (NSString *)suite {
    return @"hash";
}

- (id<CLCipher>)cipher {
    return [[Sum16Cipher alloc] init];
}

- (NSArray *)vectors {
    return [Sum16Vectors all];
}

@end
